### ISSUE TITLE
how to train a big transformer model?

### ISSUE THREAD

**Issue by rgwt123** [marian-nmt/marian-dev#267]
1.I'm trying to train a big-transformer model and just change the params( including heads from 8 to 16, dim-emb from 512 to 1024, dim-ffn from 2048 to 4096). I trained on a small zh-en corpus(about 200k), with the default params, I can get a 37.46 bleu with 1 gpu and 37.43 with 2 ,but when using the big model, it just achieve bleu 4 and then decrease to less than 1 (with 1 or 2 gpu). So if I need to change other settings to make the model converge?
2.I tried the aan to replace self-attention if the decoder part using the same corpus above, then got a bleu about 33 ,why? Have you tried it ?

**Comment by rgwt123** [comment]
@cepin19  Using lr 0.0001 can successfully train the big model on a single gpu. I think the problem is caused by a small batch_size.
I also notice a option '--optimizer-delay arg (=1)' (SGD update delay, 1 = no delay). If this option can help large mini-batch training (even on a single GPU) via delayed updates like fairseq.[https://github.com/pytorch/fairseq](url)
Thank you very much!

**Comment by cepin19** [comment]
You are welcome, and yes, I forgot about --optimizer-delay, I usually set it to 2 when I do not have many GPUs available.

**Comment by rgwt123** [comment]
@cepin19 Sorry, I still have a question. Have you tried aan? I'm using 2 different corpus, both bleu are about 3~4 lower than the model without aan.

**Comment by emjotde** [comment]
With a transformer-big configuration? I have only trained AANs with smaller models and had usually no problems. in terms of quality. The got pretty close to the original model.
